Add details to documentation on how to deploy to OpenShift and test the application (#12356)
quarkusio/quarkus: docs/src/main/asciidoc/deploying-to-openshift.adoc

* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
* Apache Maven {maven-version}
* access to an OpenShift cluster (Minishift is a viable option)

> * OpenShift CLI

Reviewer: We should probably mention that this is optional

Author: OpenShift CLI is required for running oc commands from a terminal.  A lot of tasks, not all,  could be achieved using the Web Console instead of using CLI but a developer needs to be comfortable with using CLI in my view.

Reviewer: Yeah I know, my point is that the CLI is only needed when you want to do perform the manual deployment

Author: Understood. I have updated line #22 to as follows:

* OpenShift CLI (Optional. Only required for manually deploying)

Reviewer: Looks great1

The configuration items that are available can be found in: `io.quarkus.kubernetes.deployment.OpenShiftConfig` class.
Furthermore, the items provided by `io.quarkus.deployment.ApplicationConfig` affect the OpenShift resources.

> === Build and deply (in separate steps)

Reviewer: ```suggestion
=== Build and Deploy (in separate steps)
```

Author: Thanks. Done.

> If OpenShift extension was not included during the bootstraping of the project nor added subsequently  (check pom.xml file for it), then add OpenShift as a dependency before building:

Reviewer: ```suggestion
If the OpenShift extension was not included during the bootstraping of the project nor was it added subsequently  (check pom.xml file for it), then it can be added like so:
```

Author: Thanks. Done.

quarkus.container-image.builder=jib
----

=== Build and Deploy (in a single step)

> If OpenShift extension was not included during the bootstraping of the project nor added subsequently  (check pom.xml file for it), then add OpenShift as a dependency before building:

Reviewer: ```suggestion
If the OpenShift extension was not included during the bootstraping of the project nor was it added subsequently  (check pom.xml file for it), then it can be added like so:
```

Author: Thanks. Done.

[source, subs=attributes+]
----
 ./mvnw quarkus:add-extension -Dextensions="openshift"
----

To trigger a build and deployment in a single step:

[source, subs=attributes+]
----
./mvnw clean package -Dquarkus.kubernetes.deploy=true
----

> The command above will build a jar file locally, subsequently trigger a container image build and then apply the generated OpenShift resources, right after.

Reviewer: ```suggestion
The aforementioned command will build a jar file locally, trigger a container image build and then apply the generated OpenShift resources.
```

Author: Thanks. Done.
